apps/src/main.c

```diff
@@ -1,60 +1,63 @@
-#include "stdio.h"
-#include "tcp_context.h"
 #include <signal.h>
 #include <string.h>
 
+#include "client.h"
+#include "stdio.h"
+#include "tcp_context.h"
+
 static char buffer[80];
 
-static void callback(const event_type ev, void *c_info, const void *in,
+static void callback(const event_type ev, void* client, const void* in,
                      const unsigned int len) {
-  const char *client_ip = socev_get_client_ip(c_info);
-  const unsigned short port = socev_get_client_port(c_info);
+  const char* client_ip = client_get_ip(client);
+  const unsigned short port = client_get_port(client);
   switch (ev) {
-  case CLIENT_CONNECTED:
-    printf("client connected: %s:%d\n", client_ip, port);
-    break;
-  case CLIENT_DISCONNECTED:
-    printf("client disconnected: %s:%d\n", client_ip, port);
-    break;
-  case CLIENT_DATA_RECEIVED:
-    printf("received from [%s:%d]: %s\n", client_ip, port, (const char *)in);
-    socev_callback_on_writable(c_info);
-    socev_set_timer(c_info, 1000000);
-    break;
-  case CLIENT_WRITABLE: {
-    memset(buffer, 0, sizeof(buffer));
-    int pos = sprintf(buffer, "here is your answer\n");
-    socev_write(c_info, buffer, pos);
-  } break;
-  case CLIENT_TIMER_EXPIRED: {
-    memset(buffer, 0, sizeof(buffer));
-    int pos = sprintf(buffer, "your time is up\n");
-    socev_write(c_info, buffer, pos);
-  } break;
-  default:
-    break;
+    case EVT_CLIENT_CONNECTED:
+      printf("client connected: %s:%d\n", client_ip, port);
```

Comment:
It can be %u for port because port is unsigned.

Reply:
done